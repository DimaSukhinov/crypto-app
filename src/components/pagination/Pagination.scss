.pages {
  margin: 20px 0;

  &__page {
    border: 1px solid;
    border-radius: 100%;
    padding: 5px 10px;
    margin: 0 10px;
    cursor: pointer;

    &:hover {
      background-color: #edeef0;
    }

    &-active {
      background-color: #b6b6b9;
    }
  }
}
