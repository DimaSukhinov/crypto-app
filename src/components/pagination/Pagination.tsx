import React, { useCallback } from 'react';
import './Pagination.scss';

type PaginationPropsType = {
  totalValues: number
  valuesPerPage: number
  currentPage: number
  changeCurrentPage: (page: number) => void
}

export const Pagination = React.memo(({
                                        totalValues, valuesPerPage, changeCurrentPage, currentPage,
                                      }: PaginationPropsType) => {

  const pages = [];

  for (let i = 1; i <= Math.ceil(totalValues / valuesPerPage); i++) {
    pages.push(i);
  }

  const changePage = useCallback((page: number) => () => changeCurrentPage(page), [changeCurrentPage]);

  return (
    <div className={'pages'} data-testid={'pagination'}>
      {pages.map(p => <span key={p} data-testid={'pagination-elem'}
                            className={currentPage === p ? 'pages__page pages__page-active' : 'pages__page'}
                            onClick={changePage(p)}>{p}</span>)}
    </div>
  );
});
