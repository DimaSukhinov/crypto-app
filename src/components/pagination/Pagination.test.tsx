import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Pagination } from './Pagination';

test('pagination components render', () => {
  render(
    <Pagination totalValues={100} currentPage={1} valuesPerPage={14} changeCurrentPage={jest.fn()} />,
  );

  expect(screen.getByTestId('pagination')).toBeInTheDocument();
  expect(screen.getByTestId('pagination')).toMatchSnapshot();

  expect(screen.getAllByTestId('pagination-elem')).toHaveLength(8);
  expect(screen.getByText(1)).toBeInTheDocument(); // expected number 1 on the page
  expect(screen.getByText(8)).toBeInTheDocument(); // expected number 8 on the page
});
