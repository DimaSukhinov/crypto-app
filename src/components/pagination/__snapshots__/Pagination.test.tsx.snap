// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination pagination components render 1`] = `
<div
  class="pages"
  data-testid="pagination"
>
  <span
    class="pages__page pages__page-active"
    data-testid="pagination-elem"
  >
    1
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    2
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    3
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    4
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    5
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    6
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    7
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    8
  </span>
</div>
`;

exports[`pagination components render 1`] = `
<div
  class="pages"
  data-testid="pagination"
>
  <span
    class="pages__page pages__page-active"
    data-testid="pagination-elem"
  >
    1
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    2
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    3
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    4
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    5
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    6
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    7
  </span>
  <span
    class="pages__page"
    data-testid="pagination-elem"
  >
    8
  </span>
</div>
`;
